### PR TITLE
add Apple Silicon to supporting platforms

### PR DESCRIPTION
**Pull request by ringabout**
ref https://github.com/nim-lang/Nim/issues/16211

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ architecture combinations:
 
   * Windows (Windows XP or greater) - x86 and x86_64
   * Linux (most, if not all, distributions) - x86, x86_64, ppc64 and armv6l
-  * Mac OS X (10.04 or greater) - x86, x86_64 and ppc64
+  * Mac OS X (10.04 or greater) - x86, x86_64, ppc64 and Apple Silicon
```
Comment (ringabout) [inline readme.md]: or 
```suggestion
  * Mac OS X (10.04 or greater) - x86, x86_64, ppc64 and Apple Silicon (based on the ARM64 architecture)
```

**Comment by ringabout** [comment]
CI needs to run on M1 platform

**Comment by Araq** [comment]
No, we officially support it as I have an M1 and do ensure it keeps working. Not ideal, of course, but still.

File: readme.md (hunk 1 above)
Comment (dom96) [reply readme.md]: we don't test ppc64 anymore afaik, maybe we should remove it from this list?

**Pull request merged**